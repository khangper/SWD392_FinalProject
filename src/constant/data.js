export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
