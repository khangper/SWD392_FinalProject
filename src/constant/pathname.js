export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
