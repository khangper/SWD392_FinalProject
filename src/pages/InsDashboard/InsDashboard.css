.ins-dashboard-container {
  display: flex;
  justify-content: space-between;
  flex-direction: column;
}

.ins-dashboard-top {
  width: 100%;
  background-color: #f7f7f7;
  padding-left: 5px;
}
.ins-dashboard-header {
  display: flex;
  justify-content: space-between;
  gap: 10px;
}
.ins-dashboard-header h2 {
  font-size: 20px;
  margin-top: 0;
  margin-bottom: 8px;
  font-weight: bold;
  font-family: "Roboto", sans-serif;
  color: #333;
}
.ins-dashboard-header img {
  width: 25px;
  height: 25px;
  top: 5px;
  position: relative;
}

.ins-dashboard-top .dashboard-cards {
  display: flex;
  justify-content: space-between;
  margin: 20px 0;
  padding: 10px 0;
}

.ins-dashboard-top .dashboard-card {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 3px;
  padding: 20px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 23%;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
}

.ins-dashboard-top .card-content {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.ins-dashboard-top .card-info {
  display: flex;
  flex-direction: column;
}

.ins-dashboard-top .card-info h3 {
  margin: 5px 0;
  font-size: 16px;
  color: #333;
}

.ins-dashboard-top .card-info h1 {
  margin: 10px 0;
  font-size: 24px;
  color: #333;
}

.ins-dashboard-top .new-badge {
  font-size: 12px;
  font-weight: 500;
  font-family: "Roboto", sans-serif;
  background: #efefef;
  padding: 5px 10px;
  border-radius: 3px;
  color: #fff;
  width: 70px;
  text-align: center;
}

.ins-dashboard-top .new-sales {
  background-color: #ffc136;
}

.ins-dashboard-top .new-enroll {
  background-color: #c182f9;
}

.ins-dashboard-top .new-courses {
  background-color: #ffa052;
}

.ins-dashboard-top .new-students {
  background-color: #cca1ff;
}

.ins-dashboard-top .card-icon img {
  width: 75px;
  height: 75px;
}

.ins-dashboard-left {
  flex: 0 1 65%;
  padding: 0 15px 0 20px;
  background-color: #f7f7f7;
  overflow: hidden;
  display: flex;
}

.ins-dashboard-right {
  flex: 0 1 35%;
  background-color: #f7f7f7;
}

.ins-dashboard-bottom {
  display: flex;
  justify-content: space-between;
  width: 100%;
  background-color: #f7f7f7;
}
.course-creation {
  background-color: #fff;
  padding: 20px;
  margin: 20px 0;
  display: flex;
  align-items: center;
  justify-content: space-between;
  border: 1px solid #e6e6e6;
  border-radius: 8px;
}

.course-creation-content {
  display: flex;
  align-items: center;
  width: 100%;
}

.course-creation-icon {
  width: 40px;
  height: 40px;
  margin-right: 20px;
}

.course-creation h3 {
  margin: 0;
  font-size: 18px;
  color: #333;
  flex-grow: 1;
}

.course-creation-button {
  background-color: #ff0000;
  color: #fff;
  font-size: 16px;
  padding: 10px 10px;
  border-radius: 3px;
  border: none;
  position: relative;
}

.course-creation-button:hover {
  background-color: black;
}

.course-and-news-section {
  display: flex;
  justify-content: space-between;
  gap: 10px;
  width: 100%;
}

/* Courses performance section */
.course-performance-section {
  display: flex;
  flex-direction: column;
  width: 50%;
  gap: 10px;
}

.course-performance-header {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.course-performance-title {
  font-size: 18px;
  margin: 20px 0;
}

.course-performance-btn {
  display: flex;
  gap: 5px;
}

.course-performance-btn .scroll-button {
  background-color: #fff;
  border: 1px solid #ddd;
  padding: 5px 10px;
  cursor: pointer;
  border-radius: 4px;
  position: relative;
  top: 13px;
}
.course-performance-btn .scroll-button.left {
  position: relative;
}
.course-performance-btn .scroll-button.right {
  position: relative;
}

.course-performance-btn .scroll-button:hover {
  background-color: #ff0000;
}

.course-performance-container {
  display: flex;
  overflow-x: hidden;
  scroll-behavior: smooth;
  width: 100%;
}

.course-performance-card {
  background-color: #fff;
  padding: 10px;
  border: 1px solid #efefef;
  border-radius: 3px;
  width: 100%;
  min-width: 378px;
  margin-right: 10px;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
}

.course-performance-card img {
  width: 100%;
  border-radius: 3px;
}

.course-details {
  padding: 10px 0;
}

.course-details-header {
  font-size: 14px;
  color: #999;
}
.course-performance-view {
  display: flex;
  justify-content: space-between;
}
.course-performance-view p {
  float: left;
  font-size: 14px;
  font-weight: 400;
  font-family: "Roboto", sans-serif;
  color: #686f7a;
  text-align: left;
}
.course-performance-links {
  float: left;
  width: 100%;
  margin-top: 18px;
  border-top: 1px solid #efefef;
  padding-top: 15px;
}

.course-link:first-child {
  padding-top: 0;
}

.course-link {
  padding-bottom: 5px;
  padding-top: 5px;
  font-size: 14px;
  font-family: "Roboto", sans-serif;
  display: block;
  color: #333;
  text-align: left;
  line-height: 24px;
  font-weight: bold;
  text-transform: uppercase;
}

.course-link:hover {
  text-decoration: underline;
}

.news-content {
  font-size: 14px;
  color: #999;
}

.news-link {
  display: block;
  color: #007bff;
  text-decoration: none;
  font-size: 14px;
  margin: 5px 0;
}

.news-link:hover {
  text-decoration: underline;
}

.right-section {
  width: 100%;
  display: flex;
  flex-direction: column;
  padding-left: 10px;
}

.profile-analytics-section{
display: flex;
flex-direction: column;
}
.profile-analytics-card{
  background: #fff;
    width: 100%;
    float: left;
    padding: 10px;
    border-radius: 3px;
    border: 1px solid #efefef;
    box-shadow: none;
    transition: all .2s ease-in-out;
}
.profile-analytics-title {
  font-size: 18px;
  font-weight: bold;
  color: #333;
  margin-bottom: 20px;
}

.profile-analytics-content {
  padding: 10px 5px;
    float: left;
    width: 100%;
    position: relative;
}

.current-subscribers-label {
  display: block;
  font-size: 14px;
  font-weight: 400;
  font-family: 'Roboto', sans-serif;
  color: #686f7a;
  text-align: left;
  margin-bottom: 0;
  line-height: 24px;
}

.current-subscribers {
  display: block;
  font-size: 24px;
  font-weight: 500;
  font-family: 'Roboto', sans-serif;
  color: #333;
  text-align: left;
  margin-bottom: 0;
  line-height: 24px;
  margin-top: 15px;
  padding-bottom: 20px;
  border-bottom: 1px solid #efefef;
}

.analytics-stats {
  width: 100%;
  margin-bottom: 20px;
}

.analytics-stat {
  display: flex;
  justify-content: space-between;
  width: 100%;
    padding-top: 17px;
}
.stat-left{
  float: left;
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    color: #686f7a;
    text-align: left;
}

.stat-right{
  float: right;
  font-size: 14px;
  font-weight: 400;
  font-family: 'Roboto', sans-serif;
  color: #686f7a;
  text-align: left;
}
.stat-value {
  font-size: 15px;
  font-weight: 400;
  color: #333;
}

.stat-change {
  font-size: 15px;
}

.stat-change.up {
  color: green;
}

.stat-change.down {
  color: red;
}

.profile-analytics-link {
  color: #007bff;
  text-decoration: none;
  font-size: 14px;
  margin-top: 10px;
}

.profile-analytics-link:hover {
  text-decoration: underline;
}
.sc-card{
  padding: 10px 5px;
  float: left;
  width: 100%;
  position: relative;
  background: #fff;
}
.pa-title{ 
  display: block;
  font-size: 16px;
  font-weight: bold;
  font-family: 'Roboto', sans-serif;
  color: #333;
  text-align: left;
  margin-bottom: 0;
  line-height: 26px;
}
.pa-pending-icon{
  font-size: 12px;
    font-weight: bold;
    margin-left: 10px;
    background: #ed2a26;
    color: #fff;
    padding: 5px 10px;
    border-radius: 5px;
}
.submit-course{
  font-size: 14px;
    font-family: 'Roboto', sans-serif;
    font-weight: 400;
    margin-top: 15px;
    color: #686f7a;
    font-style: italic;
    margin-bottom: 0;
}
.submit-course span{
  margin-left: 10px;
}
.delete-link{
  display: block;
    padding: 15px 0 0;
    font-size: 14px;
    font-family: 'Roboto', sans-serif;
    font-weight: 400;
    color: #333;
    text-align: left;
}
.news-link{
  display: block;
    padding: 15px 10px;
    font-size: 14px;
    font-family: 'Roboto', sans-serif;
    font-weight: 400;
    color: #333;
    text-align: left;
    border-bottom: 1px dashed #efefef;
}
.news-link:first-child{
  padding-top: 0;
}
.news-link:last-child{
  padding-bottom: 0;
    border-bottom: 0;
}