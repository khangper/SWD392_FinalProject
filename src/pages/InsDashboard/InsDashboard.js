import React, { useRef, useState } from "react";
import "./InsDashboard.css";
import dashboardIcon from "../../assets/dashboard.png";
import coursesIcon from "../../assets/BookHelp.png";
const InsDashboard = () => {
  const dashboardCards = [
    {
      title: "Total Sales",
      amount: "$350",
      newAmount: "New $50",
      icon: "https://gambolthemes.net/html-items/cursus-new-demo/images/dashboard/achievement.svg",
      newBadgeClass: "new-sales",
    },
    {
      title: "Total Enroll",
      amount: "1500",
      newAmount: "New 125",
      icon: "https://gambolthemes.net/html-items/cursus-new-demo/images/dashboard/graduation-cap.svg",
      newBadgeClass: "new-enroll",
    },
    {
      title: "Total Courses",
      amount: "130",
      newAmount: "New 5",
      icon: "https://gambolthemes.net/html-items/cursus-new-demo/images/dashboard/online-course.svg",
      newBadgeClass: "new-courses",
    },
    {
      title: "Total Students",
      amount: "2650",
      newAmount: "New 245",
      icon: "https://gambolthemes.net/html-items/cursus-new-demo/images/dashboard/knowledge.svg",
      newBadgeClass: "new-students",
    },
  ];
  const courses = [
    {
      id: 1,
      title: "Complete Python Bootcamp: Go from zero to hero in Python 3",
      views: "1.5k",
      purchased: "150",
      likes: "1k",
      comments: "875",
      reviews: "105",
      imgSrc:
        "https://gambolthemes.net/html-items/cursus-new-demo/images/courses/img-1.jpg",
      date: "First 2 days 22 hours",
    },
    {
      id: 2,
      title: "The Complete JavaScript Course 2020: Build Real Projects!",
      views: "175k",
      purchased: "1k",
      likes: "85k",
      comments: "915",
      reviews: "255",
      imgSrc:
        "https://gambolthemes.net/html-items/cursus-new-demo/images/courses/img-2.jpg",
      date: "Second 4 days 9 hours",
    },
    {
      id: 3,
      title: "Beginning C++ Programming - From Beginner to Beyond",
      views: "17k",
      purchased: "25",
      likes: "15k",
      comments: "155",
      reviews: "15",
      imgSrc:
        "https://gambolthemes.net/html-items/cursus-new-demo/images/courses/img-3.jpg",
      date: "Third 6 days 11 hours",
    },
  ];
  const newsItems = [
    {
      id: 1,
      title: "COVID-19 Updates & Resources",
      content:
        "See the latest updates to coronavirus-related content, including changes to monetization, and access new Creator support resources.",
      icon: "https://gambolthemes.net/html-items/cursus-new-demo/images/blog/img-1.jpg",
      link: "#",
    },
    {
      id: 2,
      title: "New Features for Instructors",
      content:
        "Discover the new features available to instructors, aimed at improving the teaching and learning experience.",
      icon: "https://gambolthemes.net/html-items/cursus-new-demo/images/blog/img-2.jpg",
      link: "#",
    },
    {
      id: 3,
      title: "Community Events and Meetups",
      content:
        "Join our community events and meetups to connect with other instructors and learners.",
      icon: "https://gambolthemes.net/html-items/cursus-new-demo/images/blog/img-3.jpg",
      link: "#",
    },
  ];
  const coursePerformanceRef = useRef(null);
  const newsPerformanceRef = useRef(null);
  const scrollLeftCourse = () => {
    coursePerformanceRef.current.scrollBy({ left: -410, behavior: "smooth" });
  };
  const scrollRightCourse = () => {
    coursePerformanceRef.current.scrollBy({ left: 410, behavior: "smooth" });
  };
  const scrollLeftNews = () => {
    newsPerformanceRef.current.scrollBy({ left: -410, behavior: "smooth" });
  };
  const scrollRightNews = () => {
    newsPerformanceRef.current.scrollBy({ left: 410, behavior: "smooth" });
  };

  return (
    <div className="ins-dashboard-container">
      <div className="ins-dashboard-top">
        <div className="ins-dashboard-header">
          <h2>
            <img src={dashboardIcon} alt="" /> Instructor Dashboard
          </h2>
        </div>
        <section className="dashboard-cards">
          {dashboardCards.map((card, index) => (
            <div className="dashboard-card" key={index}>
              <div className="card-content">
                <div className="card-info">
                  <h3>{card.title}</h3>
                  <h1>{card.amount}</h1>
                  <span className={`new-badge ${card.newBadgeClass}`}>
                    {card.newAmount}
                  </span>
                </div>
                <div className="card-icon">
                  <img src={card.icon} alt={`${card.title} Icon`} />
                </div>
              </div>
            </div>
          ))}
        </section>
        <section className="course-creation">
          <div className="course-creation-content">
            <img
              src={coursesIcon}
              alt="Course Creation Icon"
              className="course-creation-icon"
            />
            <h3>Jump Into Course Creation</h3>
            <button className="course-creation-button">
              Create Your Course
            </button>
          </div>
        </section>
      </div>
      <div className="ins-dashboard-bottom">
        <div className="ins-dashboard-left">
          <section className="course-and-news-section">
            <section className="course-performance-section">
              <div className="course-performance-header">
                <h2 className="course-performance-title">
                  Latest Course Performance
                </h2>
                <div className="course-performance-btn">
                  <button
                    className="scroll-button left"
                    onClick={() => scrollLeftCourse(coursePerformanceRef)}
                  ></button>
                  <button
                    className="scroll-button right"
                    onClick={() => scrollRightCourse(coursePerformanceRef)}
                  ></button>
                </div>
              </div>
              <div
                className="course-performance-container"
                ref={coursePerformanceRef}
              >
                {courses.map((course) => (
                  <div key={course.id} className="course-performance-card">
                    <a>
                      <img src={course.imgSrc} alt={course.title} />
                    </a>
                    <div className="course-details">
                      <div className="course-details-header">
                        <p className="course-view-and-date">{course.date}</p>
                      </div>
                      <a href="#" className="course-title">
                        {course.title}
                      </a>
                      <div className="course-performance-view">
                        <p>View</p> <p>{course.views}</p>
                      </div>
                      <div className="course-performance-view">
                        {" "}
                        <p>Purchased</p>
                        <p>{course.purchased}</p>
                      </div>
                      <div className="course-performance-view">
                        <p>Total Like</p> <p>{course.likes}</p>{" "}
                      </div>
                      <div className="course-performance-links">
                        <a href="#" className="course-link">
                          GO TO COURSE ANALYTICS
                        </a>
                        <a href="#" className="course-link">
                          SEE COMMENTS ({course.comments})
                        </a>
                        <a href="#" className="course-link">
                          SEE REVIEWS ({course.reviews})
                        </a>
                      </div>
                    </div>
                  </div>
                ))}
              </div>
            </section>
            <section className="course-performance-section">
              <div className="course-performance-header">
                <h2 className="course-performance-title">News</h2>
                <div className="course-performance-btn">
                  <button
                    className="scroll-button left"
                    onClick={() => scrollLeftNews(newsPerformanceRef)}
                  ></button>
                  <button
                    className="scroll-button right"
                    onClick={() => scrollRightNews(newsPerformanceRef)}
                  ></button>
                </div>
              </div>
              <div
                className="course-performance-container"
                ref={newsPerformanceRef}
              >
                {newsItems.map((news) => (
                  <div key={news.id} className="course-performance-card">
                    <a>
                      <img src={news.icon} alt={news.title} />
                    </a>
                    <div className="course-details">
                      <div className="course-details-header">
                        <p className="course-title">{news.title}</p>
                      </div>
                      <p className="news-content">{news.content}</p>
                      <div className="course-performance-links">
                        <a href={news.link} className="course-link">
                          Learn More
                        </a>
                      </div>
                    </div>
                  </div>
                ))}
              </div>
            </section>
          </section>
        </div>
        <div className="ins-dashboard-right">
          <section className="right-section">
            <section className="profile-analytics-section">
              <div className="course-performance-header">
                <h2 className="course-performance-title">Profile Analytics</h2>
              </div>
              <div className="profile-analytics-card">
                <div className="profile-analytics-content">
                  <h6 className="current-subscribers-label">
                    Current subscribers
                  </h6>
                  <h3 className="current-subscribers">856</h3>
                  <div className="analytics-stats">
                    <div className="analytics-stat">
                      <div className="stat-left"> View</div>
                      <div className="stat-right">
                        <span className="stat-value">17k</span>
                        <span className="stat-change down">▼ 75%</span>
                      </div>
                    </div>
                    <div className="analytics-stat">
                      <div className="stat-left"> Purchased (per hour)</div>
                      <div className="stat-right">
                        <span className="stat-value">1</span>
                        <span className="stat-change up">▲ 100%</span>
                      </div>
                    </div>
                    <div className="analytics-stat">
                      <div className="stat-left"> Enroll (per hour)</div>
                      <div className="stat-right">
                        <span className="stat-value">50</span>
                        <span className="stat-change down">▼ 70%</span>
                      </div>
                    </div>
                  </div>

                  <div className="course-performance-links">
                    <a href="#" className="course-link">
                      GO TO PROFILE ANALYTICS
                    </a>
                  </div>
                </div>
              </div>
            </section>
            <section className="submit-course-section">
              <div className="course-performance-header">
                <h2 className="course-performance-title">Submit Courses</h2>
              </div>
              <div className="sc-card">
                <a href="#" className="pa-title">
                  The Complete JavaScript Course 2020: Build Real Projects!
                  <span className="pa-pending-icon">Pending</span>
                </a>
                <p className="submit-course">
                  Submmited <span>1 days ago</span>
                </p>
                <a href="#" className="delete-link">
                  Delete
                </a>
              </div>
            </section>
            <section className="profile-analytics-section">
              <div className="course-performance-header">
                <h2 className="course-performance-title">
                  What's new on Cursus
                </h2>
              </div>
              <div className="sc-card">
                <a href="#" className="news-link">
                  Improved performance on Studio Dashboard
                </a>
                <a href="#" className="news-link">
                  See more Dashboard updates
                </a>
                <a href="#" className="news-link">
                  See issues-at-glance for Live
                </a>
              </div>
            </section>
          </section>
        </div>
      </div>
    </div>
  );
};

export default InsDashboard;
