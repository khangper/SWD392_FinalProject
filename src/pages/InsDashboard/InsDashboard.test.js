import React from "react";
import { render, screen } from "@testing-library/react";
import InsDashboard from "../InsDashboard/InsDashboard";
import "@testing-library/jest-dom/extend-expect";
import { renderWithProviders } from "../../constant/test-util";

describe("InsDashboard", () => {
  beforeAll(() => {
    jest.spyOn(console, "error").mockImplementation(() => {});
    jest.spyOn(console, "log").mockImplementation(() => {});
  });

  afterAll(() => {
    console.error.mockRestore();
    console.log.mockRestore();
  });
  it("renders the InsDashboard component", async () => {
    const { container } = renderWithProviders(<InsDashboard />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
