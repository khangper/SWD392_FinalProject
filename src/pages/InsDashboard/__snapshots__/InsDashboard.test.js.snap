// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsDashboard renders the InsDashboard component 1`] = `
<div>
  <div
    class="ins-dashboard-container"
  >
    <div
      class="ins-dashboard-top"
    >
      <div
        class="ins-dashboard-header"
      >
        <h2>
          <img
            alt=""
            src="dashboard.png"
          />
           Instructor Dashboard
        </h2>
      </div>
      <section
        class="dashboard-cards"
      />
      <section
        class="course-creation"
      >
        <div
          class="course-creation-content"
        >
          <img
            alt="Course Creation Icon"
            class="course-creation-icon"
            src="BookHelp.png"
          />
          <h3>
            Jump Into Course Creation
          </h3>
          <button
            class="course-creation-button"
          >
            Create Your Course
          </button>
        </div>
      </section>
    </div>
    <div
      class="ins-dashboard-bottom"
    >
      <div
        class="ins-dashboard-left"
      >
        <section
          class="course-and-news-section"
        >
          <section
            class="course-performance-section"
          >
            <div
              class="course-performance-header"
            >
              <h2
                class="course-performance-title"
              >
                Latest Course Performance
              </h2>
              <div
                class="course-performance-btn"
              >
                <button
                  class="scroll-button left"
                />
                <button
                  class="scroll-button right"
                />
              </div>
            </div>
            <div
              class="course-performance-container"
            />
          </section>
          <section
            class="course-performance-section"
          >
            <div
              class="course-performance-header"
            >
              <h2
                class="course-performance-title"
              >
                News
              </h2>
              <div
                class="course-performance-btn"
              >
                <button
                  class="scroll-button left"
                />
                <button
                  class="scroll-button right"
                />
              </div>
            </div>
            <div
              class="course-performance-container"
            />
          </section>
        </section>
      </div>
      <div
        class="ins-dashboard-right"
      >
        <section
          class="right-section"
        >
          <section
            class="profile-analytics-section"
          >
            <div
              class="course-performance-header"
            >
              <h2
                class="course-performance-title"
              >
                Profile Analytics
              </h2>
            </div>
            <div
              class="profile-analytics-card"
            >
              <div
                class="profile-analytics-content"
              >
                <h6
                  class="current-subscribers-label"
                >
                  Current subscribers
                </h6>
                <h3
                  class="current-subscribers"
                >
                  856
                </h3>
                <div
                  class="analytics-stats"
                >
                  <div
                    class="analytics-stat"
                  >
                    <div
                      class="stat-left"
                    >
                       View
                    </div>
                    <div
                      class="stat-right"
                    >
                      <span
                        class="stat-value"
                      >
                        17k
                      </span>
                      <span
                        class="stat-change down"
                      >
                        ▼ 75%
                      </span>
                    </div>
                  </div>
                  <div
                    class="analytics-stat"
                  >
                    <div
                      class="stat-left"
                    >
                       Purchased (per hour)
                    </div>
                    <div
                      class="stat-right"
                    >
                      <span
                        class="stat-value"
                      >
                        1
                      </span>
                      <span
                        class="stat-change up"
                      >
                        ▲ 100%
                      </span>
                    </div>
                  </div>
                  <div
                    class="analytics-stat"
                  >
                    <div
                      class="stat-left"
                    >
                       Enroll (per hour)
                    </div>
                    <div
                      class="stat-right"
                    >
                      <span
                        class="stat-value"
                      >
                        50
                      </span>
                      <span
                        class="stat-change down"
                      >
                        ▼ 70%
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="course-performance-links"
                >
                  <a
                    class="course-link"
                    href="#"
                  >
                    GO TO PROFILE ANALYTICS
                  </a>
                </div>
              </div>
            </div>
          </section>
          <section
            class="submit-course-section"
          >
            <div
              class="course-performance-header"
            >
              <h2
                class="course-performance-title"
              >
                Submit Courses
              </h2>
            </div>
            <div
              class="sc-card"
            >
              <a
                class="pa-title"
                href="#"
              >
                The Complete JavaScript Course 2020: Build Real Projects!
                <span
                  class="pa-pending-icon"
                >
                  Pending
                </span>
              </a>
              <p
                class="submit-course"
              >
                Submmited 
                <span>
                  1 days ago
                </span>
              </p>
              <a
                class="delete-link"
                href="#"
              >
                Delete
              </a>
            </div>
          </section>
          <section
            class="profile-analytics-section"
          >
            <div
              class="course-performance-header"
            >
              <h2
                class="course-performance-title"
              >
                What's new on Cursus
              </h2>
            </div>
            <div
              class="sc-card"
            >
              <a
                class="news-link"
                href="#"
              >
                Improved performance on Studio Dashboard
              </a>
              <a
                class="news-link"
                href="#"
              >
                See more Dashboard updates
              </a>
              <a
                class="news-link"
                href="#"
              >
                See issues-at-glance for Live
              </a>
            </div>
          </section>
        </section>
      </div>
    </div>
  </div>
</div>
`;
